src/frame.js

```diff
@@ -0,0 +1,74 @@
+function Frame() {
+  this._roll = null;
+  this._firstBall = null;
+  this._secondBall = null;
+  this._isComplete = false;
+}
+
+Frame.prototype.calculateFrameTotal = function() {
```

Comment:
I think this would resolve the problem you talked about where you wanted to get scores before the game was finished by calling `calculateFrameTotal` in `addFrame`.